### ISSUE TITLE
The blueprint using Frigate doesn't treat - and _ the same, so misses triggers unless all cameras use underscores

### ISSUE THREAD

**Issue by jonlove234** [valentinfrlch/ha-llmvision#127]
I had my frigate cameras named:
<House>-<Location>

HA device IDs replace the - with a _, which was leading to no matches.

I renamed devices in Frigate but wanted to share in case it made sense to treat -/_ the same in the lookup.

**Comment by jonlove234** [comment]
Related to the blueprint usage. I noticed I had events but they weren't
triggering the blueprint so some debugging led me to find that difference
that was the cause.

**Comment by valentinfrlch** [comment]
Thanks! This happens when taking the camera names from the mqtt message and "translating" it into the entity id in home assistant. I know what causes the issue and will take a look at it. Might take a while though.

**Comment by jonlove234** [comment]
Yeah, wonder if a regex on the camera names would help? Anyway, enjoying it
otherwise although need to work out why I'm not seeing/able to access
images!

**Comment by valentinfrlch** [comment]
Glad you enjoy the integration! For reference, this is what 'converts' Frigate's camera names into home assistant entity ids at the moment:
```
{{ trigger.payload_json['after']['camera'].replace('_', ' ')|capitalize }}
```
A regex would definitely be the better option.

**Comment by alienatedsec** [comment]
Hi @valentinfrlch 

Is there a plan for a new blueprint version to fix this?

It seems that the step below cannot be `true` because of the naming convention you discussed above.

```
value_template: |
  {% if mode == 'Frigate' %}
    {{ trigger.payload_json["type"] == "end"
       and ('camera.' + trigger.payload_json['after']['camera']|lower) in camera_entities_list
       and ((object_types_list|length) == 0 or ((trigger.payload_json['after']['label']|lower) in object_types_list))
    }}
  {%else%}
    true
  {% endif %}
```

What would be the interim solution?

@jonlove234 have you managed to amend the blueprint?

All of my cameras in Frigate (and equally MQTT) have `-` in their name.

**Comment by jonlove234** [comment]
I didn't, unfortunately. I don't have much confidence in my YAML!

**Comment by valentinfrlch** [comment]
Hi @alienatedsec, what are your camera names?
```yaml
trigger.payload_json['after']['camera']|lower) in camera_entities_list
```
This should check if the camera that triggered the frigate event is in the camera list of the blueprint as well. Which should work as long as the camera entity_ids don't have any special characters in them or have been changed.

**Comment by alienatedsec** [comment]
> Which should work as long as the camera entity_ids don't have any special characters in them or have been changed.

@valentinfrlch my cameras had `-` which is a special character preventing the automation from working properly. I had to make several changes but that was it.
